src/chainstate/burn/operations/peg_out_fulfill.rs

```diff
@@ -0,0 +1,228 @@
+        let header = test_helpers::burnchain_block_header();
+
+        let op =
+            PegOutFulfillOp::from_tx(&header, &tx).expect("Failed to construct peg-out operation");
+
+        assert_eq!(op.recipient.bytes(), recipient_address_bytes);
+        assert_eq!(op.chain_tip.as_bytes(), &chain_tip_bytes);
+        assert_eq!(op.amount, amount);
+    }
+
+    #[test]
+    fn test_parse_peg_out_fulfill_should_return_error_given_wrong_opcode() {
+        let mut rng = test_helpers::seeded_rng();
+        let opcode = Opcodes::LeaderKeyRegister;
+
+        let amount = 1;
+        let recipient_address_bytes = test_helpers::random_bytes(&mut rng);
+        let output2 = test_helpers::Output::new_as_option(amount, recipient_address_bytes);
+
+        let mut data = vec![];
+        let chain_tip_bytes: [u8; 32] = test_helpers::random_bytes(&mut rng);
+        data.extend_from_slice(&chain_tip_bytes);
+
+        let tx = test_helpers::burnchain_transaction(data, output2, opcode);
+        let header = test_helpers::burnchain_block_header();
+
+        let op = PegOutFulfillOp::from_tx(&header, &tx);
+
+        match op {
+            Err(OpError::InvalidInput) => (),
+            result => panic!("Expected OpError::InvalidInput, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_parse_peg_out_fulfill_should_return_error_given_no_second_output() {
+        let mut rng = test_helpers::seeded_rng();
+        let opcode = Opcodes::PegOutFulfill;
+
+        let output2 = None;
+
+        let mut data = vec![];
+        let chain_tip_bytes: [u8; 32] = test_helpers::random_bytes(&mut rng);
+        data.extend_from_slice(&chain_tip_bytes);
+
+        let tx = test_helpers::burnchain_transaction(data, output2, opcode);
+        let header = test_helpers::burnchain_block_header();
+
+        let op = PegOutFulfillOp::from_tx(&header, &tx);
+
+        match op {
+            Err(OpError::InvalidInput) => (),
+            result => panic!("Expected OpError::InvalidInput, got {:?}", result),
+        }
+    }
+
+    #[test]
+    fn test_parse_peg_out_fulfill_should_return_error_given_too_small_header_hash() {
+        let mut rng = test_helpers::seeded_rng();
+        let opcode = Opcodes::PegOutFulfill;
+
+        let amount = 1;
+        let recipient_address_bytes = test_helpers::random_bytes(&mut rng);
+        let output2 = test_helpers::Output::new_as_option(amount, recipient_address_bytes);
+
+        let mut data = vec![];
+        let chain_tip_bytes: [u8; 31] = test_helpers::random_bytes(&mut rng);
+        data.extend_from_slice(&chain_tip_bytes);
+
+        let tx = test_helpers::burnchain_transaction(data, output2, opcode);
+        let header = test_helpers::burnchain_block_header();
+
+        let op = PegOutFulfillOp::from_tx(&header, &tx);
+
+        match op {
+            Err(OpError::ParseError) => (),
+            result => panic!("Expected OpError::ParseError, got {:?}", result),
+        }
+    }
```

Comment:
Please add test coverage for the case where the OP_RETURN contains more than 35 bytes.

Reply:
Done :heavy_check_mark: